sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/LiveMetricsRedirectPolicy.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Monitor.OpenTelemetry.AspNetCore.Internals.LiveMetrics
+{
+    internal sealed class LiveMetricsRedirectPolicy : HttpPipelinePolicy
+    {
+        // To prevent circular redirects, max redirect is set to 1.
+        internal const int MaxRedirect = 1;
+        private string? _redirectHostValue;
+
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            ProcessAsync(message, pipeline, false).EnsureCompleted();
+        }
+
+        public override ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            return ProcessAsync(message, pipeline, true);
+        }
+
+        private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool async)
+        {
+            // Check if we've already processed the redirection
+            if (message.TryGetProperty("redirectionComplete", out object? objValue)
+                && objValue != null
+                && objValue is bool isComplete
+                && isComplete)
```

Comment:
Testing code suggestions:

```suggestion
            if (message.TryGetProperty("redirectionComplete", out object? objValue)
                && objValue is bool isComplete
                && isComplete)
```